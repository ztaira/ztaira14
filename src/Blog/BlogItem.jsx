import React, { Component } from 'react';
import Page from './Page.jsx';
import BlogActiveButtons from './BlogActiveButtons.jsx';
import './BlogItem.css';

class BlogItem extends Component {
  constructor(props) {
    super(props);
    this.TogglePost = this.TogglePost.bind(this);
    this.state = {
      active: false,
      text: 'Loading...',
    }
  }

  TogglePost() {
    this.setState({'active': !this.state.active});
  }

  render() {
    // openByDefault is necessary for testing snapshots.
    // this makes sure you can take snapshots without access to TogglePost or
    // setState or onClick
    //
    // neverOpen is necessary for testing functions
    // it's so you can create a BlogItem and easily get a reference to it without
    // having to deal with a HashRouter parent context
    if ((!this.state.active && !this.props.openByDefault) || this.props.neverOpen) {
      return (
        <a
          className="BlogItem"
          onClick={this.TogglePost}
        >
          <div className="BlogItemLeftInfo">
            <div className="BlogItemLink">
              {this.props.entry.file_name.slice(0, -3)}
            </div>
            <p className="BlogItemDescription">
              {this.props.entry.description}
            </p>
          </div>
          <p
            className="BlogItemDate">
            {this.props.entry.create_date}
          </p>
        </a>
      );
    }
    else {
      let url = 'https://raw.githubusercontent.com/ztaira14/journal/master/';
      return (
        <div className="BlogItem" >
          <BlogActiveButtons
            togglePost={this.TogglePost}
            entry={this.props.entry}
          />
          <Page
            url={url + this.props.entry.file_name}
          />
          <BlogActiveButtons
            togglePost={this.TogglePost}
            entry={this.props.entry}
          />
        </div>
      );
    }
  }
}

export default BlogItem;
