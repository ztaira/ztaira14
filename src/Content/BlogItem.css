.BlogItem {
  display: flex;
  flex-direction: row;
  padding: 20px;
  min-width: 250px;
  flex-wrap: wrap;
  text-align: left;
  margin: 0px 0px 20px 0px;
  background-color: #e4d3ff;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
  justify-content: right;
  word-wrap: break-word;
  font-family: 'Helvetica Nueue', Helvetica, Arial, sans-serif;
  color: #505050;
  transition: 0.3s;
  cursor: pointer;
}

.BlogItemLeftInfo {
  display: flex;
  flex-direction: column;
  flex-grow: 20;
  min-width: 300px;
}

.BlogItem .PageContent {
  margin: 20px 0px 20px 0px;
  padding: 0px 5% 0px 5%;
  width: 92%;
  cursor: default;
  box-shadow: none;
  max-width: 92%;
  min-width: 0px;
  background-color: #e4d3ff;
}

.BlogItemDate {
  margin: 0px;
  padding: 10px;
  display: block;
  float: right;
  flex-grow: 1;
}

.BlogItemLink {
  font-weight: bold;
  color: #505050;
  text-align: left;
  padding: 10px;
  background-color: rgba(0, 0, 0, 0);
  display: block;
  float: left;
}

.BlogItemDescription {
  margin: 0px;
  padding: 10px;
  word-wrap: break-word;
}

.BlogItem:hover {
  box-shadow: 0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.3);
}
