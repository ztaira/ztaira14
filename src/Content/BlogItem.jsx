import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Page from './Page.jsx';
import './BlogItem.css';

class BlogItem extends Component {
  constructor(props) {
    super(props);
    this.TogglePost = this.TogglePost.bind(this);
    this.state = {
      active: false,
      text: 'Loading...',
    }
  }

  TogglePost() {
    this.setState({'active': !this.state.active});
  }

  render() {
    if (!this.state.active) {
      return (
        <a
          className="BlogItem"
          onClick={this.TogglePost}
        >
          <div>
            <div className="BlogItemLink">
              {this.props.entry.file_name.slice(0, -3)}
            </div>
            <p className="BlogItemDescription">
              {this.props.entry.description}
            </p>
          </div>
          <p
            className="BlogItemDate">
            {this.props.entry.create_date}
          </p>
        </a>
      );
    }
    else {
      let url = 'https://raw.githubusercontent.com/ztaira14/journal/master/';
      return (
        <div className="BlogItem" >
          <div className="ActiveBlogButtons" >
            <a
              className="CloseButton"
              onClick={this.TogglePost}
            >
              Close
            </a>
            <Link
              className="CloseButton"
              to={"/Blog/" + this.props.entry.file_name.slice(0, -3)}
            >Go To Bookmarkable Page
            </Link>
          </div>
          <Page
            url={url + this.props.entry.file_name}
          />
          <div className="ActiveBlogButtons" >
            <a
              className="CloseButton"
              onClick={this.TogglePost}
            >
              Close
            </a>
            <Link
              className="CloseButton"
              to={"/Blog/" + this.props.entry.file_name.slice(0, -3)}
            >Go To Bookmarkable Page
            </Link>
          </div>
        </div>
      );
    }
  }
}

export default BlogItem;
