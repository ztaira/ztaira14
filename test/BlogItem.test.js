import React from 'react';
import ReactDOM from 'react-dom';
import { Route, browserHistory, HashRouter, Switch } from 'react-router-dom';
import renderer from 'react-test-renderer';
import BlogItem from '../src/Blog/BlogItem.jsx';

beforeEach(function() {
  this.entry = {
    'name': "Hello, world!",
    'file_name': "hello_world.md",
    'description': "The description, however long, goes here",
    'create_date': '2017-7-25'
  }
  this.blogItem = ReactDOM.render(
    <BlogItem entry={this.entry} openByDefault={false} neverOpen={true}/>,
    document.body.appendChild(document.createElement('div'))
  );
});

it('has not changed since the last snapshot - closed blogitem', function() {
  const blogitem = renderer.create(
    <BlogItem entry={this.entry} />
  );
  let blogitem_json = blogitem.toJSON();
  expect(blogitem_json).toMatchSnapshot();
});

it('has not changed since the last snapshot - open blogitem', function() {
  const blogitem = renderer.create(
    <HashRouter>
      <BlogItem entry={this.entry} openByDefault={true}/>
    </HashRouter>
  );
  let blogitem_json = blogitem.toJSON();
  expect(blogitem_json).toMatchSnapshot();
});

it('has a function to toggle the active state', function() {
  expect(this.blogItem.state['active']).toBe(false);
  this.blogItem.TogglePost();
  expect(this.blogItem.state['active']).toBe(true);
});

it('has some sort of loading text', function() {
  expect(this.blogItem.state['text']).toBe('Loading...');
});
