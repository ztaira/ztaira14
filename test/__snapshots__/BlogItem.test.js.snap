exports[`test has not changed since the last snapshot - closed blogitem 1`] = `
<a
  className="BlogItem"
  onClick={[Function]}>
  <div
    className="BlogItemLeftInfo">
    <div
      className="BlogItemLink">
      hello_world
    </div>
    <p
      className="BlogItemDescription">
      The description, however long, goes here
    </p>
  </div>
  <p
    className="BlogItemDate">
    2017-7-25
  </p>
</a>
`;

exports[`test has not changed since the last snapshot - open blogitem 1`] = `
<div
  className="BlogItem">
  <div
    className="BlogActiveButtons">
    <a
      className="BlogCloseButton BlogButton"
      onClick={[Function]}>
      Close
    </a>
    <a
      className="BlogLinkButton BlogButton"
      href="#/Blog/hello_world"
      onClick={[Function]}>
      Go To Bookmarkable Page
    </a>
  </div>
  <div
    className="PageContent">
    Loading
  </div>
  <div
    className="BlogActiveButtons">
    <a
      className="BlogCloseButton BlogButton"
      onClick={[Function]}>
      Close
    </a>
    <a
      className="BlogLinkButton BlogButton"
      href="#/Blog/hello_world"
      onClick={[Function]}>
      Go To Bookmarkable Page
    </a>
  </div>
</div>
`;
